src/Paprika/Store/AbandonedPage.cs

```diff
@@ -232,13 +233,13 @@ public bool TryPush(DbAddress address)
     public void AttachTail(DbAddress tail, IBatchContext batch)
     {
         Debug.Assert(page.Header.BatchId == batch.BatchId);
+        ref var abandoned = ref Data;
```

Comment:
Why ref over `Data` here? Could it be `ref Data.Next` maybe?

Also the reference is not overwritten anywhere but assigned over and over. This will destroy the chain, will it not?